tools/generator-go-sdk/internal/generator/templater_methods_discriminators_test.go

```diff
@@ -299,6 +296,18 @@ type ListCompleteResult struct {
 	Items []PandaPop
 }
 
+type ListCustomPager struct {
+	NextLink *odata.Link %[2]s
+}
+
+func (p *ListCustomPager) NextPageLink() *odata.Link {
```

Comment:
should we nil check here like in https://github.com/hashicorp/go-azure-sdk/blob/8ee56b008ba9275ff6c3282052e6b50e83073f9a/sdk/client/client_test.go#L166-L173, and should we introduce `log` package here?

Reply:
No that's mostly for testing purposes, so this is fine. If `NextPageLink` returns nil then we assume we're at the end of the dataset, so that's fine/should be logged elsewhere